tests/internal/aws_s3_local_buffer.c

```diff
@@ -0,0 +1,157 @@
+}
+
+static void test_flb_buffer_put_create_chunk()
+{
+    int ret;
+    struct flb_local_buffer *store = flb_calloc(1, sizeof(struct flb_local_buffer));
+    struct flb_output_instance *out = flb_calloc(1, sizeof(struct flb_output_instance));
+
+    store->dir = BUFFER_DIRECTORY;
+    memcpy(out->name, PLUGIN_NAME, strlen(PLUGIN_NAME));
+    store->ins = out;
+    mk_list_init(&store->chunks);
+    TEST_CHECK(mk_list_size(&store->chunks) == 0);
+
+    char *data = TEST_DATA;
+    char *key1 = KEY_1 ;
+    size_t data_len = strlen(data);
+    struct flb_local_chunk *chunk;
+
+    /* No local chunk suitable for this data has been created yet, 
+     * hence chunk should be NULL.
+     */
+    chunk = flb_chunk_get(store, key1);
+    TEST_CHECK(chunk == NULL);
+    ret = flb_buffer_put(store, chunk, key1, data, data_len);
+    TEST_CHECK(ret == 0);
+
+    /* Ensure that the data stored and retrieved from the chunk is the same. */
+    chunk = flb_chunk_get(store, key1);
+    check_chunk(chunk, key1, data);
+
+    flb_chunk_destroy(chunk);
+    flb_free(out);
+    flb_free(store);
+}
+
+static void test_flb_buffer_put_valid_chunk()
+{
+    int ret;
+    struct flb_local_buffer *store = flb_calloc(1, sizeof(struct flb_local_buffer));
+    struct flb_output_instance *out = flb_calloc(1, sizeof(struct flb_output_instance));
+
+    store->dir = BUFFER_DIRECTORY;
+    memcpy(out->name, PLUGIN_NAME, strlen(PLUGIN_NAME));
+    store->ins = out;
+    mk_list_init(&store->chunks);
+    TEST_CHECK(mk_list_size(&store->chunks) == 0); 
+
+    char *data = TEST_DATA;
+    char *key2 = KEY_2;
+    size_t data_len = strlen(data);
+    struct flb_local_chunk *chunk;
+
+    /* No local chunk suitable for this data has been created yet, 
+     * hence chunk should be NULL.
+     */
+    chunk = flb_chunk_get(store, key2);
+    TEST_CHECK(chunk == NULL);
+
+    ret = flb_buffer_put(store, chunk, key2, data, data_len);
+    TEST_CHECK(ret == 0);
+
+    /* A new chunk associated with key2 was created in the above statement, 
+     * hence this time, chunk should not be NULL. 
+     */
+    chunk = flb_chunk_get(store,key2);
+    TEST_CHECK(chunk != NULL);
+
+    chunk = flb_chunk_get(store, key2);
+    check_chunk(chunk, key2, data);
+
+    flb_chunk_destroy(chunk);
+    flb_free(out);
+    flb_free(store);
+}
+
+static void test_flb_init_local_buffer()
+{
+    int ret;
+    char *data = TEST_DATA;
+    char *key3 = KEY_3;
+    struct mk_list *tmp;
+    struct mk_list *head;
+    struct flb_local_chunk *chunk;
+    struct flb_local_buffer *store = flb_calloc(1,sizeof(struct flb_local_buffer));
+    struct flb_output_instance *out = flb_calloc(1,sizeof(struct flb_output_instance));
+    struct flb_local_buffer *new_store = flb_calloc(1, 
+                                                    sizeof(struct flb_local_buffer));
+    struct flb_output_instance *new_out= flb_calloc(1, 
+                                                    sizeof(struct flb_output_instance));
```

Comment:
Style thing: declare all of the variables first, and then allocate memory for them.

After each allocation, you should also have a check that it succeeded: ex: TEST_CHECK(store != NULL)

(In practice that should never fail ofc)

Reply:
Also, there's no real to have two output instances, you could re-use the same one for both store's

Reply:
Done